lsp-pyright.el

```diff
@@ -0,0 +1,182 @@
+  :type 'boolean
+  :group 'lsp-pyright
+  :package-version '(lsp-mode . "7.0"))
+
+;; taken from lsp-python-ms
+(defcustom lsp-pyright-python-executable-cmd "python"
+  "Command to specify the Python command for the Microsoft Python Language Server.
+Similar to the `python-shell-interpreter', but used only with mspyls.
+Useful when there are multiple python versions in system.
+e.g, there are `python2' and `python3', both in system PATH,
+and the default `python' links to python2,
+set as `python3' to let ms-pyls use python 3 environments."
+  :type 'string
+  :group 'lsp-python-ms
+  :package-version '(lsp-mode . "7.0"))
+
+;; taken from lsp-python-ms
+(defun lsp-pyright-locate-python ()
+  "Look for virtual environments local to the workspace"
+  (let* ((venv (locate-dominating-file default-directory "venv/"))
+         (sys-python (executable-find lsp-pyright-python-executable-cmd))
+         (venv-python (f-expand "venv/bin/python" venv)))
+    (cond
+     ((and venv (f-executable? venv-python)) venv-python)
+     (sys-python))))
+
+(defun lsp-pyright--begin-progress-callback (workspace &rest _)
+  (with-lsp-workspace workspace
+    (--each (lsp--workspace-buffers workspace)
+      (when (buffer-live-p it)
+        (with-current-buffer it
+          (lsp--spinner-start)))))
+  (lsp--info "Pyright language server is analyzing..."))
+
+(defun lsp-pyright--report-progress-callback (_workspace params)
+  "Log progress information."
+  (when (and (arrayp params) (> (length params) 0))
+    (lsp-log (aref params 0))))
+
+(defun lsp-pyright--end-progress-callback (workspace &rest _)
+  (with-lsp-workspace workspace
+    (--each (lsp--workspace-buffers workspace)
+      (when (buffer-live-p it)
+        (with-current-buffer it
+          (lsp--spinner-stop))))
+    (lsp--info "Pyright language server is analyzing...done")))
+
+(lsp-register-custom-settings
+ `(("pyright.disableLanguageServices" lsp-pyright-disable-language-services)
+   ("pyright.disableOrganizeImports" lsp-pyright-disable-organize-imports)
+   ("pyright.useLibraryCodeForTypes" lsp-pyright-use-library-code-for-types)
+   ("python.analysis.diagnosticMode" lsp-pyright-diagnostic-mode)
+   ("python.analysis.typeCheckingMode" lsp-pyright-typechecking-mode)
+   ("python.analysis.logLevel" lsp-pyright-log-level)
+   ("python.analysis.autoSearchPaths" lsp-pyright-auto-search-paths)
+   ("python.pythonPath" lsp-pyright-locate-python)))
+
+
+
+(lsp-dependency 'pyright
+                '(:system "pyright-langserver")
+                '(:npm :package "pyright"
+                       :path "pyright-langserver"))
+
+(lsp-register-client
+ (make-lsp-client
+  :new-connection (lsp-stdio-connection (lambda ()
+                                          (cons (lsp-package-path 'pyright)
+                                                lsp-pyright-langserver-command-args)))
+  :major-modes '(python-mode)
+  :server-id 'pyright
+  :multi-root t
+  :priority -1
```

Comment:
this and pyls priority should be different.

Reply:
As it is a new lsp-client, I set priority to lower `-2`.